File: docs-examples/hello-world-java/src/test/java/io/micronaut/graphql/docs/GraphiQLControllerTest.java
package io.micronaut.graphql.docs;

import io.micronaut.configuration.graphql.GraphiQLController;
import io.micronaut.context.BeanContext;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import jakarta.inject.Inject;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertTrue;

@MicronautTest(startApplication = false)
class GraphiQLControllerTest {

    @Inject
    BeanContext beanContext;

    @Test
    void beanOfTypeGraphiQLControllerExists() {
        assertTrue(beanContext.containsBean(GraphiQLController.class));
    }
}

File: docs-examples/hello-world-python/src/test/python/micronaut/graphql/docs/GraphiQLControllerTest.py
from typing import Annotated

from jakarta.inject import Inject
from micronaut.configuration.graphql import GraphiQLController
from micronaut.context import BeanContext
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test


@MicronautTest(startApplication=False)
class GraphiQLControllerTest:

    bean_context: Annotated[BeanContext, Inject]

    @Test
    def test_bean_of_type_graphiql_controller_exists(self) -> None:
        assert self.bean_context.containsBean(GraphiQLController)
